api/core/app/entities/app_invoke_entities.py

```diff
@@ -130,7 +130,7 @@ class AppGenerateEntity(BaseModel):
     # extra parameters, like: auto_generate_conversation_name
     extras: dict[str, Any] = Field(default_factory=dict)
 
-    # tracing instance
+    # tracing instance; use forward ref to avoid circular import at import time
     trace_manager: Optional["TraceQueueManager"] = None
```

Comment:
I was bitten by the `trace_manager` too while trying to serialize `AppGenerateEntity` and its subclasses into JSON, and converted it to a `PrivateAttr` of the `AppGenerateEntity` class.